lib/puppet/util/monkey_patches.rb

```diff
@@ -93,23 +86,4 @@ def set_default_paths
   unless OpenSSL::X509.const_defined?(:V_ERR_HOSTNAME_MISMATCH)
     OpenSSL::X509.const_set(:V_ERR_HOSTNAME_MISMATCH, 62)
   end
-
-  unless OpenSSL::X509::Name.instance_methods.include?(:to_utf8)
-    class OpenSSL::X509::Name
-      # https://github.com/openssl/openssl/blob/OpenSSL_1_1_0j/include/openssl/asn1.h#L362
-      ASN1_STRFLGS_ESC_MSB = 4
-
-      FLAGS = if RUBY_PLATFORM == 'java'
-                OpenSSL::X509::Name::RFC2253
-              else
-                OpenSSL::X509::Name::RFC2253 & ~ASN1_STRFLGS_ESC_MSB
-              end
-
-      def to_utf8
-        # https://github.com/ruby/ruby/blob/v2_5_5/ext/openssl/ossl_x509name.c#L317
-        str = to_s(FLAGS)
-        str.force_encoding(Encoding::UTF_8)
-      end
-    end
-  end
```

Comment:
Looks like the JRuby openssl still doesn't implement `#to_utf8` for `OpenSSL::X509::Name`: https://github.com/jruby/jruby-openssl/blob/master/src/main/java/org/jruby/ext/openssl/X509Name.java

Reply:
:cry:

Reply:
Do we use this in JRuby though?

Reply:
Not really, but I try to run as much of the puppet code/tests on JRuby as possible.

Reply:
Ah so I forgot that things like ACE use openssl classes and they depend on puppet, so dropping the monkey patch *could* break them if they were to call `to_utf8`. I don't think they are, but to be on the safe side I think it's best to keep the monkey patch for JRuby.

Reply:
bummer, but good to know